File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys
from collections import deque

input = sys.stdin.readline
N,M,V = map(int, input().split())
graph = [[] for _ in range(N+1)]
visited_dfs = [False] * (N+1)
visited_bfs = [False] * (N+1)

for _ in range(M):
    a,b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

def dfs(graph, start, visited):
    visited[start] = True
    print(start, end=" ")
    for v in sorted(graph[start]):
        if not visited[v]:
            dfs(graph, v, visited)

def bfs(graph, start, visited):
    q = deque([start])
    visited[start] = True
    while q:
        v = q.popleft()
        print(v, end=" ")
        for i in sorted(graph[v]):
            if not visited[i]:
                q.append(i)
                visited[i] = True 

dfs(graph, V, visited_dfs)
print()
bfs(graph, V, visited_bfs)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.swift
import Foundation

let input = readLine()!.split(separator: " ").map{Int($0)!}
let (n,m,v) = (input[0], input[1], input[2])
var dict = [Int: [Int]]()
for _ in 0..<m {
    let l = readLine()!.split(separator: " ").map{Int($0)!}
    dict[l[0], default: []].append(l[1])
    dict[l[1], default: []].append(l[0])
}

for i in 1...n {
    dict[i]?.sort()
}

var visited = Array(repeating: false, count: n+1)
func dfs(_ node: Int) {
    visited[node] = true
    print(node, terminator: " ")
    for n in dict[node, default: []] {
        if !visited[n] {
            dfs(n)
        }
    }
}

func bfs(_ node: Int) {
    var queue = [node]
    visited[node] = true
    var idx = 0
    while idx < queue.count {
        let cur = queue[idx]
        print(cur, terminator: " ")
        idx += 1
        for n in dict[cur, default: []] {
            if !visited[n] {
                visited[n] = true
                queue.append(n)
            }
        }
    }
}

dfs(v)
visited = Array(repeating: false, count: n+1)
print()
bfs(v)